src/cli-client/internals/cli/validator_test.go

```diff
@@ -0,0 +1,29 @@
+package cli_test
+
+import (
+	"testing"
+
+	"github.com/canonical/oci-factory/cli-client/internals/cli"
+)
+
+func TestValidateAndFormatDateLegalInput(t *testing.T) {
+	inputs := []string{"2006-07-08", "2008-09-10", "2024-05-01"}
+
+	for _, datetime := range inputs {
+		_, err := cli.ValidateAndFormatDate(datetime)
+		if err != nil {
+			t.Fatalf("Failed parsing legal time format yyyy-mm-dd, %v", err)
+		}
+	}
+}
+
+func TestValidateAndFormatDateBadInput(t *testing.T) {
+	inputs := []string{"2001-02-29", "2024-05-32", "01-01-2023"}
+
+	for _, datetime := range inputs {
+		_, err := cli.ValidateAndFormatDate(datetime)
+		if err == nil {
+			t.Fatalf("Parsing illegal date input [%s] didn't raise error", datetime)
+		}
+	}
+}
```

Comment:
This depends on the style but normally these two tests are combined into the same table that has an input and an error field. Then the testing code can do:
```go
for _, test := range testCases {
	_, err := cli.ValidateAndFormatDate(test.input)
	if test.err != nil {
		c.Assert(err, Equals, test.err)
		break
	}
	c.Assert(err, IsNil)
}
```

Reply:
I am sorry I wrote the code above with a `break` instead of a `continue`, that would be ideal (even though assert fails the test, if you ever change it to `c.Check` instead).